DeviceAdministration/Infrastructure/Repository/DeviceRegistryRepository.cs

```diff
@@ -213,8 +153,8 @@ public DeviceRegistryRepository(IConfigurationProvider configProvider, IDocDbRes
 
             existingDevice.DeviceProperties.HubEnabledState = isEnabled;
             existingDevice.DeviceProperties.UpdatedTime = DateTime.UtcNow;
-
-            return (await _docDbRestUtil.UpdateDocumentAsync(existingDevice)).ToObject<DeviceModel>();
+            await _documentClient.SaveAsync(deviceId, existingDevice);
```

Comment:
if this deviceId is "device.DeviceProperties.DeviceID" then you need to pass existingDevice.id here?